tests/unit/testsuite/Bolt/Boltpay/Block/Checkout/BoltpayTest.php

```diff
@@ -255,6 +289,401 @@ public function testGetSaveOrderURL()
+                "multiStepKey" => "",
+                "paymentOnlyKey" => "payOnly-key-on-homepage",
+                "routeName" => "cms",
+                "controllerName" => "index",
+                "predictedReturnValue" => "payOnly-key-on-homepage"
+            ],
+            [
+                # When only multi-step key exists on unexpected page, empty string payment-only, multi-step is used
+                "multiStepKey" => "multi+payOnly-key-on-homepage",
+                "paymentOnlyKey" => "",
+                "routeName" => "cms",
+                "controllerName" => "index",
+                "predictedReturnValue" => "multi+payOnly-key-on-homepage"
+            ],
+            [
+                # When both keys exist on product page, multi-step is used
+                "multiStepKey" => "multi+payOnly-key-on-product",
+                "paymentOnlyKey" => "payOnly+multi-key-on-product",
+                "routeName" => "catalog",
+                "controllerName" => "product",
+                "predictedReturnValue" => "multi+payOnly-key-on-product"
+            ],
+            [
+                # When only multi-step key exists on product page, null payment-only, multi-step is used
+                "multiStepKey" => "multi-key-on-product",
+                "paymentOnlyKey" => null,
+                "routeName" => "catalog",
+                "controllerName" => "product",
+                "predictedReturnValue" => "multi-key-on-product"
+            ],
+            [
+                # When only payment-only key exists on product page, null multi-step, payment-only is used
+                "multiStepKey" => null,
+                "paymentOnlyKey" => "payOnly-key-on-product",
+                "routeName" => "catalog",
+                "controllerName" => "product",
+                "predictedReturnValue" => "payOnly-key-on-product"
+            ]
+        ];
+    }
+
+    /**
+     * @test
+     * that an exception is thrown for all variants if neither a multi-step nor a payment-only key has been configured
+     *
+     * @covers ::getPublishableKeyForThisPage
+     * @dataProvider getPublishableKeyForThisPage_whenNoKeyIsConfiguredProvider
+     * @expectedException Bolt_Boltpay_BoltException
+     * @expectedExceptionMessage No publishable key has been configured.
+     *
+     * @param string|null $multiStepKey             the multi-step publishable key, empty string or null indicates not set
+     * @param string|null $paymentOnlyKey           the payment-only publishable key, empty string or null indicates not set
+     * @param string      $routeName                The magento path route name
+     * @param string      $controllerName           The magento path controller name
+     *
+     * @throws Exception if there is a problem in setting up test dependencies
+     */
+    public function getPublishableKeyForThisPage_whenNoKeyIsConfigured($multiStepKey, $paymentOnlyKey, $routeName, $controllerName)
```

Comment:
The same, use one case variable

Reply:
https://github.com/BoltApp/bolt-magento1/pull/571#discussion_r354529022

Reply:
will not change